extensions/sql-database-projects/src/projectProvider/projectProvider.ts

```diff
@@ -77,5 +77,5 @@ export class SqlDatabaseProjectProvider implements dataworkspace.IProjectProvide
 		return vscode.Uri.file(projectFile);
 	}
 
-	get providerExtensionId(): string { return GetPackageInfo()!.fullName; }
+	get providerExtensionId(): string { return getPackageInfo()!.fullName; }
```

Comment:
Explain to me again why you're adding this? This is always going to be `Microsoft.sql-database-projects` right? That will automatically always be added by the telemetry anyways in the extName column so you shouldn't need to get this yourself.

Reply:
No telemetry from _this_ (sql-database-projects) extension uses this.  The `data-workspace` extension emits a telemetry point when another extension (in this case, sql-database-projects) registers as a project/template provider, so it needs a way to ask "who goes there?" when that happens.

https://github.com/microsoft/azuredatastudio/blob/c903cd87bf81748b7643e938f4fc993e8ad106c7/extensions/data-workspace/src/common/projectProviderRegistry.ts#L26

In this case, the `extName` column is `data-workspace` (the one containing the registry), but the valuable information to capture is `sql-database-projects` (the one being registered) (or whichever other project-providing extension)

Is there already a way to get the extension identifier/id from another extension that I should be using instead?

Reply:
Ahhhhhh, yeah completely missed that. Ok that makes sense. But yeah you can just have it pull the ID of the extension from the JSON when it's registered : https://github.com/microsoft/azuredatastudio/blob/main/extensions/data-workspace/src/services/workspaceService.ts#L268

`extension.id`

Also while I was looking at this I'll say that how you appear to have the registering set up is a little odd since you require that an extension which contributes projects must have its activation exports be an IProjectProvider and nothing else.

A better pattern IMO would be to have data-workspaces expose a "registerProjectProvider" function that other extensions can call to register themselves. This decouples the extension exports from this specific implementation and also allows an extension to register multiple different providers if such a thing would ever be useful.

See https://github.com/microsoft/azuredatastudio/blob/main/extensions/resource-deployment/src/api.ts#L12 for an example of this

Reply:
@alanrenmsft, @kisantia - thoughts on what Charles' idea about the registration pattern since you two designed it?

Reply:
In any case, thanks for the tip on `extension.id`; I've reworked it to use that instead.  I never was a fan of letting the extension report its own ID, since there's no guarantee they implement that correctly.

Reply:
Charles' idea makes sense to me. @alanrenmsft is there any reason not to switch to have extensions register themselves to data-workspaces?